crates/astria-conductor/src/conductor.rs

```diff
@@ -319,6 +314,49 @@ impl Conductor {
     }
 }
 
+async fn get_executor(
+    cfg: &Config,
+    block_rx: UnboundedReceiver<crate::executor::ExecutorCommand>,
+    shutdown_rx: Receiver<()>,
+) -> eyre::Result<Executor> {
+    let hook = if cfg.enable_optimism {
+        let provider = Arc::new(
+            Provider::<Ws>::connect(cfg.ethereum_l1_url.clone())
+                .await
+                .wrap_err("failed to connect to provider")?,
+        );
+        let contract_address = Address::try_from(
```

Comment:
This is hilariously nested.

Could this be simplified by expressing this as a `block`? Or just making better use of early return or method chaining?